pkg/storage/tsdb/config.go

```diff
@@ -419,6 +420,7 @@ func (cfg *BucketStoreConfig) RegisterFlags(f *flag.FlagSet, logger log.Logger)
 	f.IntVar(&cfg.PostingOffsetsInMemSampling, "blocks-storage.bucket-store.posting-offsets-in-mem-sampling", DefaultPostingOffsetInMemorySampling, "Controls what is the ratio of postings offsets that the store will hold in memory.")
 	f.BoolVar(&cfg.IndexHeaderLazyLoadingEnabled, "blocks-storage.bucket-store.index-header-lazy-loading-enabled", true, "If enabled, store-gateway will lazy load an index-header only once required by a query.")
 	f.DurationVar(&cfg.IndexHeaderLazyLoadingIdleTimeout, "blocks-storage.bucket-store.index-header-lazy-loading-idle-timeout", 60*time.Minute, "If index-header lazy loading is enabled and this setting is > 0, the store-gateway will offload unused index-headers after 'idle timeout' inactivity.")
+	f.BoolVar(&cfg.IndexHeaderEagerLoadingStartupEnabled, "blocks-storage.bucket-store.index-header-eager-loading-startup-enabled", false, "If enabled, store-gateway will periodically persist lazy loaded index-header and loading it eagerly during startup.")
```

Comment:
```suggestion
	f.BoolVar(&cfg.IndexHeaderEagerLoadingStartupEnabled, "blocks-storage.bucket-store.index-header-eager-loading-startup-enabled", false, "If enabled, store-gateway will periodically persist lazy loaded index-headers and load them eagerly during startup.")
```

Also, is it worth mentioning here that this has no effect if lazy-loading is disabled?